pkg/v1/cli/artifact/local.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package artifact
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/common"
+)
+
+// LocalArtifact defines local artifact path
+type LocalArtifact struct {
+	Path string
+}
+
+// NewLocalArtifact creates Local Artifact object
+func NewLocalArtifact(path string) Artifact {
+	// If path is not an absolute path
+	// search under `xdg.ConfigHome/tanzu-plugin/localPath` directory
+	if !filepath.IsAbs(path) {
+		path = filepath.Join(common.DefaultLocalPluginDistroDir, "distribution", path)
+	}
+	return &LocalArtifact{
+		Path: path,
+	}
+}
+
+// Fetch an artifact.
```

Comment:
**nitpick**: The comment isn't wrong, but also doesn't feel very useful. Also, I think "fetch" is the right verb to use for the interface, but it feels like a weird verb in the context of local files, so perhaps the comment could help map the idea of "fetch" to "read"?

Something like:
```suggestion
// Fetch reads the local artifact from its path
```